service/src/main/resources/templates/index.html

```diff
@@ -0,0 +1,154 @@
+      cursor: pointer;
+    }
+
+    /* Support classes for header pinning */
+    .swagger-ui .scheme-container {
+      position: relative;
+    }
+    .swagger-ui .scheme-container.pinned {
+      position: fixed;
+      top: 0;
+      right: 0;
+      left: 0;
+      z-index: 100;
+    }
+  </style>
+</head>
+
+<body>
+<div id="swagger-ui"></div>
+<script src="/webjars/swagger-ui-dist/swagger-ui-bundle.js"> </script>
+<script src="/webjars/swagger-ui-dist/swagger-ui-standalone-preset.js"> </script>
+<script th:inline="javascript">
+
+   // Adds support for pinning the auth bar when the user scrolls down far enough to hide the bar
+    const pinLoginPlugin = function(system) {
+      return {
+        afterLoad: function(system) {
+          var offsetY;
+          var authBar;
+          document.addEventListener('scroll', function() {
+            if (offsetY === undefined) {
+              // Note: the auth bar is not a React component, so we can't use the standard plugin approach to modify
+              var authBars = document.getElementsByClassName('scheme-container');
+              if (authBars.length > 0) {
+                authBar = authBars[0];
+                offsetY = authBar.offsetTop;
+              }
+            }
+            if (window.scrollY > offsetY) {
+              authBar.classList.add('pinned');
+            } else {
+              authBar.classList.remove('pinned');
+            }
+          });
+        }
+      }
+    }
+
+    // Removes the online validation since some code gen bugs cause us to have some specs that cause the validator to squawk
+  const clearValidator = function(system) {
+    return {
+      components: {
+        onlineValidatorBadge: function() { return null; },
+      }
+    };
+  }
+
+  window.onload = function() {
+    // Begin Swagger UI call region
+    const ui = SwaggerUIBundle({
+      url: 'openapi.yml',
+      dom_id: '#swagger-ui',
+      deepLinking: true,
+      presets: [
+        SwaggerUIBundle.presets.apis,
+        SwaggerUIStandalonePreset
+      ],
+      plugins: [
+        SwaggerUIBundle.plugins.DownloadUrl,
+        pinLoginPlugin,
+        clearValidator
+      ],
+      layout: 'StandaloneLayout',
+      displayOperationId: true,
+      docExpansion: 'none',
+      defaultModelsExpandDepth: -1, // hide the huge list of schemas under the routes
+      defaultModelExpandDepth: 2, // affects the schema shown for a request or response
+      oauth2RedirectUrl: window.location.protocol + '//' + window.location.host + '/webjars/swagger-ui-dist/oauth2-redirect.html',
+      tagsSorter: function (a, b) {
+        // Sort function to ensure that upper case tags show up first (e.g. ASCII sort)
+        if (a === b) {
+          return 0;
+        } else {
+          return a < b ? -1 : 1;
+        }
+      }
+    })
+    // End Swagger UI call region
+
+    ui.initOAuth({
+      clientId: [[${clientId}]],
+      scopes: "openid email profile"
```

Comment:
I wonder if there's a way to get the scopes from the openapi yaml instead of hardcoding them here?

Reply:
from what I understand from html and and the`/webjars/swagger-ui-dist/swagger-ui.css`, I think the page is coded to look for the clientId value and I would need to mess with the webjar to put it into scopes I think.